python/dgl/heterograph.py

```diff
@@ -1854,31 +2447,71 @@ def __contains__(self, vid):
+        """Whether the graph has some particular node(s) of a given type.
 
         Parameters
         ----------
-        vid : int, iterable, tensor
-            Node ID(s).
+        vid : node ID(s)
+            The node ID(s) for query. The allowed formats are:
+
+            - ``int``: The ID of a single node.
+            - ``Tensor``: A 1D tensor that contains the IDs of multiple nodes, whose data type and
+              device should be separately the same as the idtype and device of the graph.
+            - ``iterable[int]``: A sequence (e.g. list, tuple, numpy.ndarray)
+              of integers that contains the IDs of multiple nodes.
         ntype : str, optional
-            The node type. Can be omitted if there is only one node type
-            in the graph. (Default: None)
+            The node type for query. It is required if the graph has
+            multiple node types.
 
         Returns
         -------
         bool or bool Tensor
-            Each element is a bool flag, which is True if the node exists,
-            and is False otherwise.
+
+            - If :attr:`vid` is an ``int``, the result will be a ``bool`` indicating
+              whether the graph has the particular node.
+            - If :attr:`vid` is a 1D ``Tensor`` or ``iterable[int]`` of node IDs,
+              the result will be a bool Tensor whose i-th element indicates whether
+              the graph has node :attr:`vid[i]` of the given type.
 
         Examples
         --------
+
+        The following example uses PyTorch backend.
+
+        >>> import dgl
+        >>> import torch
+
+        Create a graph with two node types -- 'user' and 'game'.
+
+        >>> g = dgl.heterograph({
+        >>>     ('user', 'follows', 'user'): (torch.tensor([0, 1]), torch.tensor([1, 2])),
+        >>>     ('user', 'plays', 'game'): (torch.tensor([3, 4]), torch.tensor([0, 1]))
+        >>> })
+
+        Query for the nodes.
+
         >>> g.has_nodes(0, 'user')
         True
-        >>> g.has_nodes(4, 'user')
+        >>> g.has_nodes(3, 'game')
         False
-        >>> g.has_nodes([0, 1, 2, 3, 4], 'user')
-        tensor([True, True, True, False, False])
+        >>> g.has_nodes(torch.tensor([3, 0, 1]), 'game')
+        tensor([False,  True,  True])
         """
+        if not isinstance(vid, (numbers.Integral, Iterable)):
+            raise DGLError('Expect an int, a tensor or a sequence for vid, '
+                           'got {}'.format(type(vid)))
+
+        if isinstance(vid, numbers.Integral) and vid < 0:
+            raise DGLError('Expect a non-negative value for vid, got {:d}'.format(vid))
+
+        if isinstance(vid, Iterable) and not F.is_tensor(vid):
+            # NaN/Inf values cannot appear in int32/int64 tensors
+            utils.detect_nan_in_iterable(vid, 'vid')
+            utils.detect_inf_in_iterable(vid, 'vid')
+
+        if isinstance(vid, Iterable):
+            utils.assert_nonnegative_iterable(vid, 'vid')
```

Comment:
jeez ... to many of this kinds of checks. Simplify it with the tensor check as in the `dgl.graph` API.